### PR TITLE
Address syntax errors in configuration + refactor

### PR DESCRIPTION
**Pull request by flands**
This change was primarily driven by the following syntax errors:

* enable_traces in Stackdriver needs to be enable_tracing
* jaeger receiver needs to be jaeger:

As part of this change, refactor and cleanup up readme a bit. Also
ensure consistent formatting in all sections. Changes include:

* Added deployment section
* Abstracted configuration section out of agent
* Abstracted E2E example out of agent
* Moved advanced topics to separate pages
* Updated diagrams

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,51 +139,21 @@ exporters:
 ```
 
 #### <a name="agent-config-receivers"></a>Receivers
-Agent provides a couple of receivers that receive spans from instrumentation libraries.
 
-#### <a name="details-receivers-opencensus"></a>OpenCensus
+One ore more receivers can be configured. An example of all available receivers is provided below.
+By default, the opencensus receiver is enabled on the OpenCensus Agent.
 
-This receiver receives spans from OpenCensus instrumented applications and translates them into the internal span types that
```
Comment (odeke-em) [inline README.md]: Sorry, why was this removed?

File: README.md [hunk 2]
```diff
@@ -141,51 +139,21 @@ exporters:
...
 receivers:
     opencensus:
         address: "127.0.0.1:55678"
-```
-
-By default this receiver is ALWAYS started since it is the point of the "OpenCensus agent"
-
-#### <a name="details-receivers-zipkin"></a>Zipkin
 
-This receiver receives spans from Zipkin "/v2" API HTTP uploads and translates them into the internal span types that are then
```
Comment (odeke-em) [inline README.md]: Please revert this change.

File: README.md [hunk 3]
```diff
@@ -141,51 +139,21 @@ exporters:
...
-Its address can be configured in the YAML configuration file under section "receivers", subsection "zipkin" and field "address".
-
-For example:
-```yaml
-receivers:
     zipkin:
         address: "localhost:9411"
-```
-
-#### <a name="details-receivers-jaeger"></a>Jaeger
```
Comment (odeke-em) [inline README.md]: Please revert this change.

File: README.md [hunk 4]
```diff
@@ -141,51 +139,21 @@ exporters:
...
-
-This receiver receives spans from Jaeger collector HTTP and Thrift uploads and translates them into the internal span types that are then
-sent to the collector/exporters.
-
-Its address can be configured in the YAML configuration file under section "receivers", subsection "jaeger" and fields "collector_http_port", "collector_thrift_port".
```
Comment (odeke-em) [inline README.md]: Also broad change.

**Comment by flands** [comment]
@odeke-em the receiver change was for consistency -- happy to go the other way as well. For exporters they are listed in one section, for receivers they are not. Not sure the value of taking up a ton of readme space to break them out individually, but happy to switch back if you think it is better.
